### PR TITLE
Issue #7315: LocalFileSystem Glob FileExists

### PR DESCRIPTION
**Pull request by hawkfish**
is_last_chunk should always trigger existence tests, not just when crawling.

### CODE REVIEW

File: src/common/local_file_system.cpp [hunk 1]
```diff
@@ -989,7 +989,7 @@ vector<string> LocalFileSystem::Glob(const string &path, FileOpener *opener) {
 			if (previous_directories.empty()) {
 				result.push_back(splits[i]);
 			} else {
-				if (recursive_search && is_last_chunk) {
+				if (recursive_search || is_last_chunk) {
```
Comment (Mytherin) [inline src/common/local_file_system.cpp:992]: This should likely be done whenever there was a glob before - not only for the last chunk. For example could we add a test case with a glob like `test/*/dir/*.parquet` (with a constant directory in the middle) and a directory structure like this:

```sql
a/dir/x.parquet
b/dir2/z.parquet
c/dir/y.parquet
```

Comment (Mytherin) [reply src/common/local_file_system.cpp:992]: On second thought I think this should just be `is_last_chunk`

Comment (Mytherin) [reply src/common/local_file_system.cpp]: Yes I think it should already work - but it is not just about working but also about minimizing the number of calls to `FileExists` and `DirectoryExists` (which can be expensive, particularly when many are executed). `recursive_search || is_last_chunk` means we are always calling `FileExists` and `DirectoryExists` when recursive_search is true.

**Review by samansmink — APPROVED** [review]
Looks good!

**Pull request merged**